### ISSUE TITLE
Unhandled exception

### ISSUE THREAD

**Issue by rlbisbe** [Keboo/PebbleSharp#1]
Using the PebbleSharp sample, trying to connect to my pebble from my Windows Phone 8.1 (Lumia 1520). I'm having the following exception:

Element not found. (Exception from HRESULT: 0x80070490)

The exception rises on the PebbleScanner:

```
  public async Task OpenAsync()
    {
        try
        {
            await _socket.ConnectAsync( _PeerInformation.HostName, "1" );
            _streamWatcher = new StreamWatcher( _socket.InputStream );
            _streamWatcher.DataAvailible += StreamWatcherOnDataAvailible;
        }
        catch ( Exception e )
        {
            Debug.WriteLine( e.ToString() );
        }
    }
```

For more information, I have a Pebble with firmware 2.6

Any idea on this?

BTW, if you need help with the project, I'd be happy to contribute ;)

**Comment by edwige-seminara** [comment]
Hi,

I have the same exception with my phone, please do you have a solution to fix that ?
Regards,

**Comment by Keboo** [comment]
Not at this time. I have finally free'd up some time to address the issue that are preventing the library from working with the latest firmware. after those are addressed I will take another stab at getting the WP demo app working.

**Comment by brookpatten** [comment]
I know this is an old issue but I thought I would comment here in case you come back to it.  I have a fork of PebbleSharp that I'm working on specifically for running on Mono/Linux.  While that part may or may not be of interest to you, I have done a lot of work to PebbleSharp.Core too to get things working with the latest firmware and bundle layout, as well as adding appmessage support.  If you're interested I could clean it up and send a PR for just those bits.

**Comment by Novo** [comment]
@brookpatten can you make your fork public? This repo looks dead ...

**Comment by brookpatten** [comment]
@Novo It is public, https://github.com/brookpatten/PebbleSharp.Mono.BlueZ

**Comment by Keboo** [comment]
@brookpatten Yes I would happily accept pull requests. I will admit since my pebble watch broke I have not spent much/any time working on this. I would also be willing to give people contributor access if there is interest moving this library forward.

**Comment by brookpatten** [comment]
Ok, I'll be happy to put that together.  My fork has some changes in it that are overly invasive in the context of a PR so I'll sort through that and submit a PR to this project.

I will likely be working on it for the forseable future as I have another project that depends on it being up-to-date with the latest pebble release.  If you don't have the interest in it any longer I'd be happy to maintain it.

BTW I feel your pain on the hardware, I'm on my pebble #2 and it seems to be dying as well.
